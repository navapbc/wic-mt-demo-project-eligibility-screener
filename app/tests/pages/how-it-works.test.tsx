import { render } from '@testing-library/react'

import HowItWorks from '@pages/how-it-works'

import { setup } from '../helpers/setup'
import {
  testAccessibility,
  testActionButtonRoute,
} from '../helpers/sharedTests'

/**
 * Test setup
 */

const route = '/how-it-works'
const backRoute = '/'
const forwardRoute = '/eligibility'

/**
 * Begin tests
 */

it('should match full page snapshot', () => {
  setup(route)
  const { container } = render(
    <HowItWorks backRoute={backRoute} forwardRoute={forwardRoute} />
  )
  expect(container).toMatchSnapshot()
})

it('should pass accessibility scan', async () => {
  setup(route)
  await testAccessibility(
    <HowItWorks backRoute={backRoute} forwardRoute={forwardRoute} />
  )
})

it('should have an action button that routes to forwardRoute', async () => {
  const { mockSession, user } = setup(route)
  const element = (
    <HowItWorks backRoute={backRoute} forwardRoute={forwardRoute} />
  )
  await testActionButtonRoute(element, forwardRoute, 'Check', user)
})
