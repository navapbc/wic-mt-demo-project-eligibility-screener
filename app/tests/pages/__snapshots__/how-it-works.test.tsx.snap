// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match full page snapshot 1`] = `
<div>
  <a
    class="usa-link"
    href="/"
  >
    Back
  </a>
  <h1>
    How it works
  </h1>
  <ol
    class="usa-process-list"
  >
    <li
      class="usa-process-list__item"
    >
      <h2
        class="usa-process-list__heading"
      >
        Check your eligibility
      </h2>
      <p
        class="margin-top-1"
      >
        You can start applying for WIC by checking to see if you're eligible.
      </p>
    </li>
    <li
      class="usa-process-list__item"
    >
      <h2
        class="usa-process-list__heading"
      >
        WIC clinic staff will schedule a certification appointment
      </h2>
      <p
        class="margin-top-1"
      >
        <strong>
          If you are eligible
        </strong>
        , a staff member from your local WIC office will call you for additional information and to set up a certification appointment.
      </p>
    </li>
    <li
      class="usa-process-list__item"
    >
      <h2
        class="usa-process-list__heading"
      >
        Complete your appointment with your local WIC clinic
      </h2>
      <p
        class="margin-top-1"
      >
        During your appointment, you'll need to show documentation of your income, identity, and residence.
      </p>
    </li>
  </ol>
  <div
    class="usa-alert usa-alert--warning usa-alert--no-icon"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        <strong>
          Note:
        </strong>
         this form does not guarantee enrollment in WIC; your eligibility will be confirmed when you have your first appointment with WIC.
      </p>
    </div>
  </div>
  <button
    class="usa-button usa-button--small display-block margin-top-6"
  >
    Check your eligibility
  </button>
</div>
`;
